src/state.cairo

```diff
@@ -127,19 +97,20 @@ pub struct TxIn {
     /// The reference to the previous output that is being used as an input.
     pub previous_output: OutPoint,
     /// The witness data for transactions.
-    pub witness: @ByteArray
+    /// A list of items (of different size) pushed onto stack before sig script execution.
+    pub witness: Span<ByteArray>,
 }
 
-
 /// A reference to a transaction output.
 #[derive(Drop, Copy)]
 pub struct OutPoint {
```

Comment:
How `OutPoint` is related to `UtreexoOutput` and `UtreexoProof`?

Reply:
Or `Outpoint` should become `UtreexoOutput`?

Reply:
`OutPoint` contains unique output ID (txid + vout) and metadata required to resolve the output.

`UtreexoOutput` contains output ID, coinbase specific metadata, and the output itself.

Although they do overlap, there is no particular connection. Maybe it makes sense to rename `UtreexoOutput` to `UtreexoLeaf` to avoid confusion.

Reply:
`OutPoint` is now reused as utreexo leaf